### PR TITLE
Added libfl.

### PR DESCRIPTION
**Pull request by kogut**
Please add the following dependency to the rosdep database.

## Package name:

libfl-dev

## Package Upstream Source:

https://github.com/westes/flex

## Purpose of using this:

This lexical analyzer generator is generally useful as a development library.

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/buster/libfl-dev

- Ubuntu: https://packages.ubuntu.com/
   - https://packages.ubuntu.com/focal/libfl-dev
   
- Fedora: https://src.fedoraproject.org/browse/projects/
  - NOT AVAILABLE
- Arch: https://www.archlinux.org/packages/
  - NOT AVAILABLE
- Gentoo: https://packages.gentoo.org/
  - NOT AVAILABLE
- macOS: https://formulae.brew.sh/
  - NOT AVAILABLE
- Alpine: https://pkgs.alpinelinux.org/packages
  - NOT AVAILABLE

### CODE REVIEW

File: rosdep/base.yaml [hunk 1]
```diff
@@ -2871,6 +2871,9 @@ libfftw3:
   nixos: [fftw, fftwSinglePrec]
   openembedded: [fftw@meta-oe]
   ubuntu: [libfftw3-3, libfftw3-dev]
+libfl-dev:
+  debian: [libfl-dev]
```
Comment (cottsay) [inline rosdep/base.yaml:2875]: Looks like the Fedora package underwent some naming changes, but it does exist: https://src.fedoraproject.org/rpms/flex#bodhi_updates

The automation is also indicating that openSUSE has this package: https://software.opensuse.org/package/flex

```suggestion
  debian: [libfl-dev]
  fedora:
    '*': [flex-devel]
    '33': [libfl-devel]
  opensuse: [libfl-devel]
```

**Comment by kogut** [comment]
Thank you for the improvement, @cottsay .

File: rosdep/base.yaml (hunk 1 above)
Comment (jacobperron) [reply rosdep/base.yaml:2875]: @cottsay I'm curious, why single out 33? It looks like rpms are available for Fedora 34 and 35 as well.

Comment (cottsay) [reply rosdep/base.yaml:2875]: The package name changed in 34. In 33 and earlier, it was called `flex-devel`, and was completely renamed to `libfl-devel` for Fedora 34 and later (covered by the `*`): https://src.fedoraproject.org/rpms/flex/c/9adf2296f3603e2ee68328283ad625d080ae1c2a?branch=rawhide

Comment (cottsay) [reply rosdep/base.yaml:2875]: Also: Fedora 33 is the oldest non-EOL Fedora release right now.

Comment (jacobperron) [reply rosdep/base.yaml:2875]: Thanks, I didn't get that from the source package overview page!

**Review by jacobperron — APPROVED** [review]

**Review by wjwwood — APPROVED** [review]

**Pull request merged**